### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by breznak** [htm-community/htm.cuda#1]
Add to Readme 
- [x] requirements
- [x] steps how to build
- [ ] high level doc of this work
  - [x] differences to c++ version/limitations (only local inhibition), forced separation of cols into "blocks"(how many,...)
  - [ ] hi-lev What does it do - CUDA-wise (loads data in batch, execute in parallel each "block",...) 
- [ ] comments in SpatialPooler.cu, HelloSP.cu - why there are "duplicit" methods (getPermanences), document cuda impl a bit
- [x] mention performance gains
- [x] Notes on Future work (what else can be optimized (in SP, in HTM,..)

**Comment by breznak** [comment]
fixed in #4
